File: skiCentar/skiCentar.Model/DailyWeather.cs
using System;

namespace skiCentar.Model
{
    public class DailyWeather
    {
        public int Id { get; set; }

        public DateTime? Date { get; set; }

        public decimal? Temperature { get; set; }

        public decimal? Precipitation { get; set; }

        public decimal? WindSpeed { get; set; }

        public decimal? Humidity { get; set; }

        public string? WeatherCondition { get; set; }

        public decimal? SnowHeight { get; set; }

        public int? ResortId { get; set; }

        public virtual Resort? Resort { get; set; }
    }
}

File: skiCentar/UI/skicentar_mobile/lib/models/daily_weather.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:skicentar_mobile/models/resort.dart';

part 'daily_weather.g.dart';

@JsonSerializable()
class DailyWeather {
  int? id;
  DateTime? date;
  int? resortId;
  Resort? resort;
  double? temperature;
  double? precipitation;
  double? windSpeed;
  double? humidity;
  double? snowHeight;
  String? weatherCondition;

  DailyWeather({this.id, this.date,this.resortId,this.temperature,this.precipitation,this.windSpeed,this.humidity,this.snowHeight,this.weatherCondition});

  factory DailyWeather.fromJson(Map<String,dynamic> json) => _$DailyWeatherFromJson(json);

  Map<String,dynamic> toJson() => _$DailyWeatherToJson(this);
}
